File: frameworks/python/starlette/Client/Kiota/middleware/middleware_request_builder.py
from __future__ import annotations
from collections.abc import Callable
from kiota_abstractions.base_request_builder import BaseRequestBuilder
from kiota_abstractions.get_path_parameters import get_path_parameters
from kiota_abstractions.request_adapter import RequestAdapter
from typing import Any, Optional, TYPE_CHECKING, Union

if TYPE_CHECKING:
    from .four.four_request_builder import FourRequestBuilder
    from .none_.none_request_builder import NoneRequestBuilder
    from .sixteen.sixteen_request_builder import SixteenRequestBuilder

class MiddlewareRequestBuilder(BaseRequestBuilder):
    """
    Builds and executes requests for operations under /middleware
    """
    def __init__(self,request_adapter: RequestAdapter, path_parameters: Union[str, dict[str, Any]]) -> None:
        """
        Instantiates a new MiddlewareRequestBuilder and sets the default values.
        param path_parameters: The raw url or the url-template parameters for the request.
        param request_adapter: The request adapter to use to execute the requests.
        Returns: None
        """
        super().__init__(request_adapter, "{+baseurl}/middleware", path_parameters)
    
    @property
    def four(self) -> FourRequestBuilder:
        """
        The four property
        """
        from .four.four_request_builder import FourRequestBuilder

        return FourRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def none_(self) -> NoneRequestBuilder:
        """
        The none property
        """
        from .none_.none_request_builder import NoneRequestBuilder

        return NoneRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def sixteen(self) -> SixteenRequestBuilder:
        """
        The sixteen property
        """
        from .sixteen.sixteen_request_builder import SixteenRequestBuilder

        return SixteenRequestBuilder(self.request_adapter, self.path_parameters)
    

File: frameworks/java/ktor/Client/Kiota/client/kiota/middleware/MiddlewareRequestBuilder.java
package client.kiota.middleware;

import client.kiota.middleware.four.FourRequestBuilder;
import client.kiota.middleware.none.NoneRequestBuilder;
import client.kiota.middleware.sixteen.SixteenRequestBuilder;
import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.RequestAdapter;
import java.util.HashMap;
import java.util.Objects;
/**
 * Builds and executes requests for operations under /middleware
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class MiddlewareRequestBuilder extends BaseRequestBuilder {
    /**
     * The four property
     * @return a {@link FourRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public FourRequestBuilder four() {
        return new FourRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The none property
     * @return a {@link NoneRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public NoneRequestBuilder none() {
        return new NoneRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The sixteen property
     * @return a {@link SixteenRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public SixteenRequestBuilder sixteen() {
        return new SixteenRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * Instantiates a new {@link MiddlewareRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public MiddlewareRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/middleware", pathParameters);
    }
    /**
     * Instantiates a new {@link MiddlewareRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public MiddlewareRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/middleware", rawUrl);
    }
}
